carbon-design-system/carbon
# feat(upgrade): adds featureflag-deprecate-flags-prop codemod and deprecates flags prop

Closes https://github.com/carbon-design-system/carbon/issues/17243

New props to feature flag props were added here https://github.com/carbon-design-system/carbon/pull/17239 

This introduces a new codemod to Convert FeatureFlags' `flags` object prop to individual `boolean` props
and  `deprecates` the existing `flags` prop
#### Changelog

**New**

- Added a new `featureflag-deprecate-flags-prop` codemod 
- Added `test fixtures` for all existing featureflags and tests to validate 
- `flags` prop for FeatureFlag is marked as deprecated 


**Changed**

- Updated `FeatureFlags/overview.stories.mdx` to document the codemod 
- Added instructions for running the `codemod` to the flags deprecation warning message


#### Testing / Reviewing

- Ensure that the test fixtures make sense - input is what someones code should look like before running the codemod, and the output is what we expect the codemod to generate/change it to.
- Make sure the docs read well and I didn't miss anything
- I used below command to run the codemode locally to verify it works  ( --parser=tsx used to process typescript files)
- `jscodeshift -t featureflag-deprecate-flags-prop.js --parser=tsx   _your_file_path_`

curious to hear how you feel about this
